### PR TITLE
ETCM-972: Add skeleton for future ExecutionSync - fetching part

### PR DESCRIPTION
**Pull request by AnastasiiaL**
# Description

Added a skeleton for a future solution

### CODE REVIEW

File: src/main/scala/io/iohk/ethereum/blockchain/sync/regular/ChainManagerService.scala [hunk 1]
```diff
@@ -0,0 +1,28 @@
+package io.iohk.ethereum.blockchain.sync.regular
+
+import cats.data.NonEmptyList
+import io.iohk.ethereum.blockchain.sync.regular.BlockImporter.BlockImportType
+import io.iohk.ethereum.domain.Block
+import io.iohk.ethereum.network.Peer
+import monix.eval.Task
+
+//not used atm, a part of the future ExecutionSync
+object ChainManagerService {
```
Comment (leo-bogastry) [inline src/main/scala/io/iohk/ethereum/blockchain/sync/regular/ChainManagerService.scala:10]: Is the idea to merge this PR as is, and have this modified in ETCM-1005 (Design new sync layer and chains manager) or later?

Comment (AnastasiiaL) [reply src/main/scala/io/iohk/ethereum/blockchain/sync/regular/ChainManagerService.scala:10]: yes, ETCM-1005 is an independent effort

**Review by leo-bogastry — APPROVED** [review]

**Review by AurelienRichez — APPROVED** [review]

File: src/main/scala/io/iohk/ethereum/blockchain/sync/regular/ChainManagerService.scala [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+  type ValidationError
+  type Branch
+
+  //saves blocks to the storage
+  def processBlocksFromFetcher(peer: Peer): Task[Either[ValidationError, Branch]] = ???
+
+  //saves blocks to the storage
+  def processMinedBlocks(): Task[Either[ValidationError, Branch]] = ???
```
Comment (strauss-m) [inline src/main/scala/io/iohk/ethereum/blockchain/sync/regular/ChainManagerService.scala:19]: what is the reason to process a fetched and a mined block differently ?

Comment (AnastasiiaL) [reply src/main/scala/io/iohk/ethereum/blockchain/sync/regular/ChainManagerService.scala:19]: they are coming from different paths: fetched blocks are coming from the BlockFetcher, mined blocks are coming from the controller and mined blocks can't fit the queue of blocks formed by a BlockFetcher that was fetching an explicit range

**Pull request closed without merging**